src/charm.py

```diff
@@ -121,24 +70,116 @@ def _on_update_status(self, event):
 
+    ##############################################
+    #               PROPERTIES                   #
+    ##############################################
+    @property
+    def mysql(self) -> MySQL:
+        """Returns MySQL object"""
+        peers_data = self.model.get_relation("mysql").data[self.app]
+        mysql_config = {
+            "app_name": self.model.app.name,
+            "host": self.unit_ip,
+            "port": self.model.config["port"],
+            "user_name": "root",
+            "mysql_root_password": peers_data["mysql_root_password"],
+        }
+        return MySQL(mysql_config)
+
+    @property
+    def unit_ip(self) -> str:
+        """Returns unit's IP"""
+        return str(self.model.get_binding(PEER).network.bind_address)
+
+    @property
+    def env_config(self) -> dict:
+        """Return the env_config for pebble layer"""
+        config = self.model.config
+        peers_data = self.model.get_relation("mysql").data[self.app]
+        env_config = {}
+        env_config["MYSQL_ROOT_PASSWORD"] = peers_data["mysql_root_password"]
+
+        if config.get("MYSQL_USER") and config.get("MYSQL_PASSWORD"):
+            env_config["MYSQL_USER"] = config["MYSQL_USER"]
+            env_config["MYSQL_PASSWORD"] = config["MYSQL_PASSWORD"]
+
+        if config.get("MYSQL_DATABASE"):
+            env_config["MYSQL_DATABASE"] = config["MYSQL_DATABASE"]
+
+        return env_config
+
+    ##############################################
+    #             UTILITY METHODS                #
+    ##############################################
+    def _mysql_root_password(self) -> str:
+        """
+        Returns MYSQL_ROOT_PASSWORD from the config,
+        if the password isn't in StoredState, generates one.
+        """
+        password_from_config = self.config["MYSQL_ROOT_PASSWORD"]
+        if password_from_config:
+            logger.debug("Retriving MYSQL_ROOT_PASSWORD from config")
+            return password_from_config
+        else:
+            logger.debug("MYSQL_ROOT_PASSWORD generated")
+            return MySQL.new_password(20)
+
+    def _update_peers(self):
+        if self.unit.is_leader():
+            peers_data = self.model.get_relation("mysql").data[self.app]
+
+            if not peers_data.get("mysql_root_password"):
+                peers_data["mysql_root_password"] = self._mysql_root_password()
+
+    def _configure_pod(self):
+        """Configure the Pebble layer for MySQL."""
+        if not self._stored.pebble_ready:
+            self.unit.status = MaintenanceStatus(
+                "Waiting for Pod startup to complete"
+            )
+            return False
+
+        layer = self._build_pebble_layer()
+
+        if not layer["services"]["mysql"]["environment"].get(
+            "MYSQL_ROOT_PASSWORD", False
+        ):
+            self.unit.status = MaintenanceStatus(
+                "Awaiting leader node to set MYSQL_ROOT_PASSWORD"
+            )
+            return False
```

Comment:
I did not see `container.add_layer`

Reply:
Was in another method. Now it is in the same place.